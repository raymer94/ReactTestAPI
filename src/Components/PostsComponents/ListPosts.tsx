import { PostsModelsDTO } from "./Posts.models";
import * as ReactBootstrap from 'react-bootstrap';
import { useHistory } from "react-router-dom";

export default function ListPosts(props: ListPostsProps) {

    const history = useHistory();

    const routeChange = (postId: any) =>{ 
      let path = `CommentsPosts/${postId}`; 
      history.push(path);
    }

    return (
        <>
            <h3>Post's List</h3>    
                <ReactBootstrap.Table striped bordered hover>
                    <thead className="">
                        <tr>
                            <th>UserId</th>
                            <th>Id</th>
                            <th>Title</th>
                            <th>Body</th>
                        </tr>
                    </thead>
                    <tbody>

                        {props.model?.map(val =>
                            <tr style={{cursor: "pointer"}} key={val.id} onClick={() => routeChange(val.id)}>
                                <td>{val.userId}</td>
                                <td>{val.id}</td>
                                <td>{val.title}</td>
                                <td>{val.body}</td>
                            </tr>
                        )}
                    </tbody>
                </ReactBootstrap.Table>          
        </>
    )
}

interface ListPostsProps {
    model?: PostsModelsDTO[]
}