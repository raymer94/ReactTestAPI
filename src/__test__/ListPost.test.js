import React from 'react';
import renderer from 'react-test-renderer';
import ListPosts from '../Components/PostsComponents/ListPosts';

test('List Snapshot test', () => {
  const component = renderer.create(
    <ListPosts/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});