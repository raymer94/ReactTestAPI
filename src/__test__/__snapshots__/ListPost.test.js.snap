// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Array [
  <h3>
    Post's List
  </h3>,
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead
      className=""
    >
      <tr>
        <th>
          UserId
        </th>
        <th>
          Id
        </th>
        <th>
          Title
        </th>
        <th>
          Body
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;

exports[`List Snapshot test 1`] = `
Array [
  <h3>
    Post's List
  </h3>,
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead
      className=""
    >
      <tr>
        <th>
          UserId
        </th>
        <th>
          Id
        </th>
        <th>
          Title
        </th>
        <th>
          Body
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
